wrappers/internal/service_unit_gen.go

```diff
@@ -71,6 +71,16 @@ func generateServiceNames(snap *snap.Info, appNames []string) []string {
 	return names
 }
 
+func ensureMinSystemdDuration(in timeout.Timeout, field string) time.Duration {
+	// If 0, the field is not included in the unit, and the default systemd value is used
+	inDur := time.Duration(in)
+	if inDur > 0 && inDur < time.Microsecond {
+		logger.Noticef("Warning: %s set to 1us as it cannot be smaller than that", field)
```

Comment:
The only concern I have is that this is largely invisible unless one goes looking into the logs. We could issue a warning, but this involves messing with the state which isn't available here. Maybe you could leave a TODO?